### ISSUE TITLE
ctrl-o creates duplicate items in Windows

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#13240]
### Describe the bug

In Windows, make sure your buffer_editor is set and hit ctrl+o. For me, I have it set to nvim. Now paste something in to nvim and save it, and exit.

I pasted in this 'repeat command' today and when I saved and exited, I get this on the prompt. It what I put in the editor twice instead of just once.
![image](https://github.com/nushell/nushell/assets/343840/cd0085a7-4223-497a-9cce-9605d35e4448)

### How to reproduce

See above

### Expected behavior

Only show the text once.

### Configuration

|key|value|
|-|-|
|version|0.95.1|
|major|0|
|minor|95|
|patch|1|
|branch|main|
|commit_hash|b679c2bfa255e6153c5d962cbd541d7a63ba3cab|
|build_os|windows-x86_64|
|build_target|x86_64-pc-windows-msvc|
|rust_version|rustc 1.77.2 (25ef9e3d8 2024-04-09)|
|rust_channel|1.77.2-x86_64-pc-windows-msvc|
|cargo_version|cargo 1.77.2 (e52e36006 2024-03-26)|
|build_time|2024-06-26 07:44:40 -05:00|
|build_rust_channel|release|
|allocator|mimalloc|
|features|default, sqlite, system-clipboard, trash|
|installed_plugins|formats 0.95.1, gstat 0.95.1, inc 0.95.1, jwalk 0.5.0, polars 0.95.1, query 0.95.1, stress_internals 0.95.1|

**Comment by sholderbach** [comment]
Twice as in shown twice (display bug) or twice as in twice in the buffer to be executed (logic bug for dealing with the file)

**Comment by fdncred** [comment]
I just reproduced it again. This is how it works today on the latest main.
1. with a blank prompt hit ctrl+o and put this text in nvim
```nushell
def "repeat command" [
    command: closure

    --sleep: duration = 1sec
] {
    loop {
            clear
            do $command | print
            sleep $sleep
    }
```
3. in nvim :wq
4. in nushell you see 1 copy of the pasted text
5. hit enter to submit it
6. you now see 2 copies of the pasted text

**Comment by fdncred** [comment]
Here's a gif example of pasting the below text in nvim using ctrl-o. (i manually pause in a few places so you can see what's going on)
Steps:
1. clear screen
2. ctrl-o
3. paste buffer
4. save in nvim (ignore nvim error)
5. notice how it shows up on the repl correctly
6. hit enter and see how it eats the last row of pasted text, kind of pastes over it and executes it, showing the output.
```nushell
"one two three\none two three" | lines
| str replace -r '^\S* (.*)' '$1'
| to json
```
![ctrlo](https://github.com/user-attachments/assets/b53f8fd0-e688-4884-9239-ab8bd272ede8)

**Comment by sholderbach** [comment]
OK so a display bug (what is the start location of the current prompt) and not a bug affecting the code that gets executed.

**Comment by fdncred** [comment]
Ya, I think you're right @sholderbach. I had to watch my gif over and over again. I did notice that there were not 2 separate outputs as well but overwrote the last line on enter.

**Comment by sholderbach** [comment]
I mean we could rule it out with something that has other side effects, but that we get the prompt location wrong is much more likely based on previous experience ^^

**Comment by ArjunCodess** [comment]
hi, i'd like to work on this.

the extra line is the prompt start row. when ctrl-o returns on a prompt row at a different column, the painter stays on that row. the text ctrl-o inserts is unchanged.

the patch is in reedline. is that the repo you want it on?

**Comment by fdncred** [comment]
ya, I think so but @kronberger-droid can decide

**Comment by kronberger-droid** [comment]
Yeah its already open on reedline, but we did some changes there already. Did you check on main already? @ArjunCodess

**Comment by ArjunCodess** [comment]
i checked reedline main. select_prompt_row still moves down a row when the column is above 0. the open reedline pull request stays on that row when the cursor is still inside the old prompt range.

**Comment by ArjunCodess** [comment]
@kronberger-droid i checked reedline main after #1201. that change reuses a bottom-row prompt only when the cursor comes back on the same cell. ctrl-o that inserts text moves the cursor, so can_reuse_prompt_at is false, and select_prompt_row still starts one row lower when the column is above 0.

https://github.com/nushell/reedline/pull/1227 only covers that remaining case. the cursor is still inside the old prompt rows, on a different column. it does not change what ctrl-o inserts.
